### PR TITLE
Delete the WAIT task in the generator of the training dataset.

### PR DESCRIPTION
**Pull request by workingloong**
The logic is complex to create the dataset for the training loop. We don't need waiting tasks in the generator of the dataset.

### CODE REVIEW

File: elasticdl/python/master/task_dispatcher.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def check_exceed_max_task_retries(self, task):
 
     def finished(self):
         """Return if all tasks are done"""
-        return all([not self._todo, not self._eval_todo, not self._doing])
```
Comment (brightcoder01) [inline elasticdl/python/master/task_dispatcher.py:368]: Will we have `_eval_todo` anymore?

File: elasticdl/python/worker/task_data_service.py [hunk 2]
```diff
@@ -169,69 +171,24 @@ def get_dataset(self):
...
-                    logger.info("No tasks for now, maybe more later")
-                    # There are too many requests to get task from the master
-                    # if the worker does not sleep.
-                    time.sleep(5)
-                else:
-                    logger.info("No more task, stopping")
                 break
             with self._lock:
                 if task.type == elasticdl_pb2.TRAIN_END_CALLBACK:
                     self._pending_train_end_callback_task = task
```
Comment (brightcoder01) [inline elasticdl/python/worker/task_data_service.py:190]: Will we have more than one TRAIN_END_CALLBACK tasks?
If we have break here, will we miss the other train end callback tasks?

Comment (workingloong) [reply elasticdl/python/worker/task_data_service.py:190]: We will not have other tasks after the training.

**Review by skydoorkai — COMMENTED** [review]
gltm as long as training tests are passed.

**Review by skydoorkai — APPROVED** [review]

**Review by brightcoder01 — APPROVED** [review]

**Pull request merged**